homeassistant/components/switchbot/cover.py

```diff
@@ -0,0 +1,146 @@
+    )
+    _attr_assumed_state = True
+
+    def __init__(
+        self,
+        coordinator: SwitchbotDataUpdateCoordinator,
+        idx: str | None,
+        mac: str,
+        name: str,
+        password: str,
+        retry_count: int,
+    ) -> None:
+        """Initialize the Switchbot."""
+        super().__init__(coordinator, idx, mac, name)
+        self._attr_unique_id = idx
+        self._device = self.coordinator.switchbot_api.SwitchbotCurtain(
+            mac=mac, password=password, retry_count=retry_count
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
+        last_state = await self.async_get_last_state()
+        if not last_state or ATTR_CURRENT_POSITION not in last_state.attributes:
+            return
+
+        self._attr_current_cover_position = last_state.attributes[ATTR_CURRENT_POSITION]
+        self._last_run_success = last_state.attributes["last_run_success"]
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return the state attributes."""
+        return {**super().extra_state_attributes}
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return if the cover is closed."""
+        return self.coordinator.data[self._idx]["data"]["position"] <= 20
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the curtain."""
+
+        _LOGGER.info("Switchbot to open curtain %s", self._mac)
+
+        async with self.coordinator.api_lock:
+            self._last_run_success = bool(
+                await self.hass.async_add_executor_job(self._device.open)
+            )
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the curtain."""
+
+        _LOGGER.info("Switchbot to close the curtain %s", self._mac)
+
+        async with self.coordinator.api_lock:
+            self._last_run_success = bool(
+                await self.hass.async_add_executor_job(self._device.close)
+            )
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the moving of this device."""
+
+        _LOGGER.info("Switchbot to stop %s", self._mac)
+
+        async with self.coordinator.api_lock:
+            self._last_run_success = bool(
+                await self.hass.async_add_executor_job(self._device.stop)
+            )
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Move the cover shutter to a specific position."""
+        position = kwargs.get(ATTR_POSITION)
+
+        _LOGGER.info("Switchbot to move at %d %s", position, self._mac)
+
+        async with self.coordinator.api_lock:
+            self._last_run_success = bool(
+                await self.hass.async_add_executor_job(
+                    self._device.set_position, position
+                )
+            )
+
+    @property
+    def current_cover_position(self) -> int:
```

Comment:
Implementing this will prevent the shorthand `self._attr_current_cover_position` used above from working because it will override how its being derived.